dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java

```diff
@@ -5981,4 +5983,1268 @@ public void requiredQualdropTest() throws Exception {
+                                          .withName("Parent Community")
+                                          .build();
+
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunity)
+                                           .withName("Collection 1")
+                                           .build();
+
+        WorkspaceItem witem = WorkspaceItemBuilder.createWorkspaceItem(context, col1)
+                                                  .withTitle("Example Title")
+                                                  .withIssueDate("2019-11-21")
+                                                  .withSubject("ExtraEntry")
+                                                  .build();
+
+        ResourcePolicyBuilder.createResourcePolicy(context)
+                             .withDspaceObject(witem.getItem())
+                             .withPolicyType(TYPE_CUSTOM)
+                             .withName("openaccess")
+                             .build();
+
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+
+        getClient(tokenAdmin).perform(get("/api/submission/workspaceitems/" + witem.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.sections.defaultAC.discoverable", is(true)))
+                             .andExpect(jsonPath("$.sections.defaultAC.accessConditions[0].name",
+                                              is("openaccess")))
+                             .andExpect(jsonPath("$.sections.defaultAC.accessConditions[1].name").doesNotExist());
+
+        List<Operation> addAccessCondition = new ArrayList<Operation>();
+        Map<String, String> accessCondition = new HashMap<String, String>();
+        accessCondition.put("name", "openaccess");
+        addAccessCondition.add(new AddOperation("/sections/defaultAC/accessConditions/-", accessCondition));
+
+        String patchBody = getPatchContent(addAccessCondition);
+        getClient(tokenAdmin).perform(patch("/api/submission/workspaceitems/" + witem.getID())
+                             .content(patchBody)
+                             .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                             .andExpect(status().isUnprocessableEntity());
+
+        getClient(tokenAdmin).perform(get("/api/submission/workspaceitems/" + witem.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.sections.defaultAC.discoverable", is(true)))
+                             .andExpect(jsonPath("$.sections.defaultAC.accessConditions[0].name",
+                                              is("openaccess")))
+                             .andExpect(jsonPath("$.sections.defaultAC.accessConditions[1].name").doesNotExist());
+    }
+
+    @Test
+    public void patchAddAccesConditionReplaceCompletelyTest() throws Exception {
```

Comment:
Since there's a duplicate check when adding an access condition, can you also add a test to verify you can't replace the access conditions completely with 2 the same values?
The code did make me doubt that was caught

Reply:
The duplication check is indeed improper. It was not requested by the contract and also after consultation with @abollini  it is not appropriate. It would be similar to the situation where an user add two times the same author: the REST will not reject the request